File: 1.TimeComplexity/python/timeComplexity3.py
# Reverse a list 

# O(N) Complexity
a = [1,56,89,40,45,89]

c = a[::-1]
print(c)


# O(N) Complexity but slightly better.
n = len(a)
for i in range(0,int(n/2)):
    temp = a[i]
    a[i] = a[n-i-1]
    a[n-i-1] = temp
print(a)

File: 1.TimeComplexity/typescript/root/timeComplexity3.ts



const a:number[] =  [1,56,89,40,45,89]
// a.reverse()
// console.log(a)

const n = a.length

for (let i:number=0;i<n/2;i++){
    const temp = a[i]
    a[i] = a[n-i-1]
    a[n-i-1] = temp
}
console.log(a)


